docker/README.md

```diff
@@ -0,0 +1,148 @@
+# Docker and exercism.io
+
+Optionally, one can develop (and even deploy) exercism.io using a tool
+called Docker. The primary advantage of using it is that you don't have to
+install a ton of dependencies--like a ruby version manager, ruby, postgres,
+nodejs, and a bunch of Ruby gems--on your own system. The disadvantage,
+however, is that you may need to acquaint yourself with Docker, and you
+need a 64-bit machine.
+
+For a basic overview, see [What is Docker?][] on the Docker website.
+
+**Note:** At the present time, developing the front-end via Docker
+isn't working, but this will hopefully be remedied soon.
+
+## Installing Docker
+
+### Linux
+
+First, you'll want to install [Docker Engine][] if you haven't already.
+
+Then install Docker Compose:
+
+    curl -L https://github.com/docker/compose/releases/download/1.5.2/docker-compose-`uname -s`-`uname -m` > /usr/local/bin/docker-compose
+    chmod +x /usr/local/bin/docker-compose
+
+### Other Platforms
+
+Please install [Docker Toolbox][]. This will ensure that you have
+Docker, Docker Machine, and Docker Compose on your system.
+
+Finally, make sure all future work with this project is done via the
+**Docker Quickstart Terminal**, *not* your standard terminal or
+command prompt. Using this special terminal is critical because it
+sets up some environment variables and services that are required for
+Docker to work on your platform.
+
+Note that on your operating system, the hard work is all being done
+behind-the-scenes by a Linux-based virtual machine. The software used
+to communicate between your OS and the VM is called [Docker Machine][].
+This indirection can sometimes cause confusion and difficulties, but
+this document will try to address potential issues when they may arise.
+
+**Note to Windows Users:** Due to various Windows issues, it's 
+probably easiest to develop by setting up a Linux virtual machine
+using a tool like [VirtualBox][], and following the Docker instructions
+for Linux, rather than using Docker Toolbox on Windows. However, you're
+welcome to give it a shot!
+
+## Setting up exercism.io
+
+First, you'll want to set up GitHub OAuth as per the instructions in
+[`CONTRIBUTING.md`][]. Create an `.env` file in the root of your
+repository with your GitHub OAuth credentials:
+
+```
+EXTERNAL_PORT=4567
+EXERCISM_GITHUB_CLIENT_ID=your_git_client_id
+EXERCISM_GITHUB_CLIENT_SECRET=your_git_client_secret
```

Comment:
I was confused about the `git client` thing in these placeholder values, since my git client is what I use on the command-line to talk to git, not anything to do with github. How about your-github-oauth-client-id and your-github-oauth-client-secret?